server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1717,7 +1717,6 @@ public void revokeEntitlements(List<Entitlement> entsToRevoke, Set<String> alrea
             int entQuantity = ent.getQuantity() != null ? ent.getQuantity() : 0;
 
             pool.setConsumed(pool.getConsumed() - entQuantity);
-            pool.getEntitlements().remove(ent);
```

Comment:
After further consideration, we've opted not to continue with explicit deletion queries here, as the lack of runtime consistency combined with the complexity of the revoke/refresh operation itself potentially opens the door to hard-to-debug consistency and querying issues further down the path. We'll consider revisiting direct queries again in the future if this area continues to be a performance bottleneck.